packages/marshal/src/helpers/remotable.js

```diff
@@ -77,6 +77,18 @@ const checkRemotableProtoOf = (original, check = x => x) => {
    *        }}
    */
   const proto = getPrototypeOf(original);
+  const protoProto = getPrototypeOf(proto);
+  if (
+    typeof original === 'object' &&
+    proto !== objectPrototype &&
+    protoProto !== objectPrototype
+  ) {
+    return (
+      // eslint-disable-next-line no-use-before-define
+      RemotableHelper.canBeValid(proto, check) && checkRemotable(proto, check)
```

Comment:
I do intend the arbitrary depth recursion, even though agoric-sdk currently uses only one or two. The reason is that we're explaining FarClasses as class-like, making things more intuitive for those used to programming with classes. It would be natural for them to expect to be able to define FarClasses that inherit from other FarClasses. The new internal representation in this PR could naturally support that, and we may want to provide that.

Having experienced the coordination difficulties of agoric-sdk needing endo to expand its remotable recognition rules (the need to patch. the breakage of CI because of something that doesn't see the patch), I thought I'd get ahead of the game and prepare to accept that deeper recursion. AFAICT, there's no reason not to, and it keep options open both for us, and for others who want to build new representations on top of endo.

Resolving.